Sources/TypeChecker/TypeChecker+Expressions.swift

```diff
@@ -130,6 +132,42 @@ extension TypeChecker {
                                       expression: argument.expression))
         }
       }
+    } else if case .matchedFunction(let matchingFunction) =
+      environment.matchFunctionCall(functionCall,
+                                    enclosingType: functionCall.identifier.enclosingType ?? enclosingType,
+                                    typeStates: typeStates,
+                                    callerProtections: callerProtections,
+                                    scopeContext: passContext.scopeContext!) {
+
+      if let externalCall = passContext.externalCall {
+        // check value parameter (type)
+        if matchingFunction.declaration.isPayable {
+          if let valueParameter: FunctionArgument = externalCall.getHyperParameter(parameterName: "value") {
+            let parameterType = environment.type(of: valueParameter.expression,
+              enclosingType: enclosingType,
+              typeStates: typeStates,
+              callerProtections: callerProtections,
+              scopeContext: passContext.scopeContext!)
+
+            if parameterType != .userDefinedType("Wei") {
```

Comment:
You should really do `!parameterType.isCurrencyType` to accommodate multiple currencies and to reduce duplication of the `"Wei"` constant.

Reply:
Alternatively for just Wei, `"Wei"` = `RawType.StdlibType.wei.rawValue`